c7n/resources/cw.py

```diff
@@ -315,3 +316,72 @@ def process_resource_set(self, client, accounts, resources):
             if found:
                 results.append(r)
         return results
+
+
+@LogGroup.action_registry.register('set-encryption')
+class EncryptLogGroup(BaseAction):
+    """Encrypt/Decrypt a log group
+
+    :example:
+
+    .. code-block: yaml
+
+        policies:
+          - name: encrypt-log-group
+            resource: log-group
+            filters:
+              - kmsKeyId: absent
+            actions:
+              - type: set-encryption
+                kms_key_id: kms:key:arn
+                state: True
+
+          - name: decrypt-log-group
+            resource: log-group
+            filters:
+              - kmsKeyId: kms:key:arn
+            actions:
+              - type: set-encryption
+                state: False
+    """
+    schema = type_schema(
+        'set-encryption',
+        kms_key_id={'type': 'string'},
```

Comment:
rather than two separate options, just using kms-key and then doing the key id lookup if its not an arn would simplify the end user options.